src/contoroller/category/category.js

```diff
@@ -78,13 +65,13 @@ class CategoryController {
   }
 
   //** OK **//
-  async getSome(req, res) {
-    const someCategories = await this.Category.findAll({
-      where: { type: req.params.type, deletedAt: null },
+  async getAll(req, res) {
+    const allCategories = await this.Category.findAll({
+      where: { userId: req.user.id, type: req.query.type, deletedAt: null },
```

Comment:
What if `req.query.type` is null?

You need to check for the query param first and if it exists add it to your where clause.

````
const whereClause = { userId: req.user.id, deletedAt: null };

if  (req.query.type) {
  whereCluase.type = req.query.type;
}

````

Reply:
done